src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,102 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService regServ;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        regServ = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("someLogin");
+        user.setPassword("SomePassword_1234!");
+        user.setAge(21);
+    }
+
+    @Test
+    void register_emptyUser_NotOk() {
+        Assertions.assertThrows(RuntimeException.class, () -> regServ.register(null));
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(RuntimeException.class, () -> regServ.register(user));
+    }
+
+    @Test
+    void register_emptyLogin_NotOk() {
+        user.setLogin("");
+        Assertions.assertThrows(RuntimeException.class, () -> regServ.register(user));
+    }
+
+    @Test
+    void register_existingUser_NotOk() {
+        StorageDao object = new StorageDaoImpl();
```

Comment:
move it to method with @BeforeAll annotation, change name, create private field StorageDao